tests/base/test_stats_utils.py

```diff
@@ -281,3 +281,210 @@ def test_logsumexp_edge_b():
     ary = np.array([1.0, 2.0, 3.0])
     assert _logsumexp(ary, b=0) == -np.inf
     assert _logsumexp(ary, b_inv=0) == np.inf
+
+
+@pytest.mark.parametrize(
+    "shape,axis,expected_shape",
+    [
+        ((4, 500, 20), (0, 1), (20,)),
+        ((4, 500, 1), (0, 1), (1,)),
+        ((4, 500, 8, 10), (0, 1), (8, 10)),
+        ((4, 500, 20), (-3, -2), (20,)),
+    ],
+)
+def test_logsumexp_loo_shapes(rng, shape, axis, expected_shape):
+    log_lik = rng.normal(loc=-5, scale=2, size=shape)
+    n_samples = shape[0] * shape[1]
+
+    scipy_result = logsumexp(log_lik, axis=axis, b=1 / n_samples)
+    arviz_result = _logsumexp(log_lik, axis=axis, b=1 / n_samples)
+
+    assert scipy_result.shape == expected_shape
+    assert arviz_result.shape == expected_shape
+    assert_array_almost_equal(scipy_result, arviz_result, decimal=10)
+
+    max_log_lik = np.max(log_lik, axis=axis)
+    assert np.all(scipy_result <= max_log_lik + 1)
+    assert np.all(scipy_result >= -10)
```

Comment:
```suggestion
    assert np.all(scipy_result <= max_log_lik + 1e-6)
    assert np.all(scipy_result >= max_log_lik - np.log(n_samples)  - 1e-6)
```
We may want to use tighter bounds. But also, we are already testing value per value against scipy, so probably not that important.